### PR TITLE
Feature: sync cache

### PR DESCRIPTION
**Pull request by vadymmarkov**
`Cache` is born to be async, so I've decided that there is no place for sync functions in `CacheAware` protocol, it kinda breaks the whole architecture of the library. But if for some reason you really need to perform cache operations synchronously (on your own risk 😄), there are helpers for that: 

**SyncHybridCache** for `HybridCache`
```swift
let cache = HybridCache(name: "Mix")
let syncCache = SyncHybridCache(cache)

// Add UIImage to cache synchronously
syncCache.add("image", object: UIImage(named: "image.png"))

// Retrieve image from cache synchronously
let image: UIImage? = syncCache.object("image")

// Remove an object from the cache
syncCache.remove("image")

// Clean the cache
syncCache.clear()
```

**HybridCache** for type safe `Cache`
```swift
let cache = Cache<UIImage>(name: "ImageCache")
let syncCache = SyncCache(cache)

syncCache.add("image", object: UIImage(named: "image.png"))
let image = syncCache.object("image")
syncCache.remove("image")
syncCache.clear()
```

This PR also adds source code documentation and this is nice 📖 

Fixes https://github.com/hyperoslo/Cache/issues/33

### CODE REVIEW

**Comment by vadymmarkov** [comment]
Travis is as always 😬

File: README.md [hunk 1]
```diff
@@ -158,6 +161,43 @@ cache.remove("image")
 cache.clear()
 ```
 
+### SyncHybridCache
+
+`Cache` is born to be async, but if for some reason you need to perform cache
+operations synchronously, there is a helper for that.
```
Comment (zenangst) [inline README.md]: I think it should read; 

`Cache` was born to be async.

**Comment by zenangst** [comment]
Travis...

**Comment by arden** [comment]
great

**Pull request merged**